packages/server/src/tests/header/headerSelection.ts

```diff
@@ -97,35 +93,56 @@ const supportersTest: HeaderTest = {
     ],
 };
 
-const getNonSupportersTest = (edition: Edition): HeaderTest => {
-    if (edition === 'UK') {
-        return nonSupportersTestUK;
-    } else if (edition === 'US') {
-        return nonSupportersTestUS();
-    }
-    return nonSupportersTestNonUK;
-};
+const hardcodedTests = [supportersTest, nonSupportersTestUK, nonSupportersTestNonUK];
 
-export const selectHeaderTest = (
+// Exported for Jest testing
+export const selectBestTest = (
     targeting: HeaderTargeting,
-): Promise<HeaderTestSelection | null> => {
-    const select = (): HeaderTest => {
-        if (targeting.showSupportMessaging) {
-            return getNonSupportersTest(targeting.edition);
-        } else {
-            return supportersTest;
+    allTests: HeaderTest[],
+): HeaderTestSelection | null => {
+    const { showSupportMessaging, countryCode } = targeting;
+
+    const selectedTest: HeaderTest | undefined = allTests.find(test => {
+        const { isOn, userCohort, locations } = test;
+
+        if (!isOn) {
+            return false;
         }
-    };
 
-    const test = select();
-    const variant = test.variants[targeting.mvtId % test.variants.length];
-    if (test && variant) {
-        return Promise.resolve({
-            test,
-            variant,
-            moduleName: header.name,
-            modulePathBuilder: variant.modulePathBuilder,
-        });
+        if (!audienceMatches(showSupportMessaging, userCohort)) {
+            return false;
+        }
+
+        const testCountryCode = countryCode != null ? countryCode.toUpperCase() : '';
+        if (!inCountryGroups(testCountryCode, locations)) {
+            return false;
+        }
+
+        // Need an extra test here to check if user is in AB variant? - userIsInTest()
```

Comment:
we can add `userIsInTest` in a follow up PR. It's actually for limiting the % of browsers which are assigned to a test, and not related to variants

Reply:
Will take out all the comments I left re userIsInTest() - I didn't understand how it played with the code we're using to assign users to variants (in ab file)